### ISSUE TITLE
Validate array of objects - first or second attribute is required, but never both

### ISSUE THREAD

**Issue by bluemanos** [express-validator/express-validator#705]
I have array of objects:
```json
	"materials": [
		{ "material_id": 7, "percentage":9, "pi":33  },
		{ "material_id": 9, "percentage":19, "pi":50  },
		{ "composition_id": 8, "percentage":25, "pi":99 }
	]
```
My validation problem is that the `material_id` or `compositon_id` is required. Bot never both! If one appears, then the other one is optional. And when both appears then I should throw an error.

```        check('materials').exists().isArray(),
        check('materials.*.material_id').isInt(),
        check('materials.*.composition_id').isInt(),
        check('materials.*.percentage').exists().isInt(),
        check('materials.*.pi').exists().isInt(),
```
How to write the custom rule?

**Comment by Axort** [comment]
I believe you can do something like:

`check('materials.*.material_id').if(check('materials.*.composition_id').not().exists()).exists().withMessage('Theres no composition_id so material_id is required')`

`check('materials.*.composition_id').if(check('materials.*.material_id').not().exists()).exists().withMessage('Theres no material_id so composition_id is required')`

Probably you should put this validations in a oneOf().

Hope this helps.
